src/core/tokenize/match.ts

```diff
@@ -146,13 +153,37 @@ export function _matchGrammar (
 				}
 
 				tokenList.removeRange(removeFrom, removeCount);
+				let byGroups = match.groups ? tokenizeByNamedGroups(match) : null;
+
+				if (byGroups?.length > 1) {
+					content = byGroups.map(arg => {
+						let content = typeof arg === 'string' ? arg : arg.content;
+						let type = typeof arg === 'string' ? undefined : arg.type;
+
+						if (insideGrammar) {
+							let localInsideGrammar = insideGrammar[type] ?? insideGrammar;
+
+							if (typeof localInsideGrammar === 'function') {
+								// Late resolving
+								localInsideGrammar = resolve.call(
+									prism,
+									localInsideGrammar(match.groups)
+								);
+							}
+
+							content = tokenize.call(prism, content, localInsideGrammar);
+						}
+
+						return typeof arg === 'object' && arg.type
```

Comment:
Why not `type` instead of `arg.type` here and in the following line?

Reply:
TS was complaining